applications/exporter/application_round_result_exporter.py

```diff
@@ -0,0 +1,194 @@
+
+    def _get_header_rows(self) -> ApplicationSectionExportRow:
+        return ApplicationSectionExportRow(
+            application_id="hakemuksen numero",
+            application_status="hakemuksen tila",
+            applicant="hakija",
+            section_id="hakemusosan numero",
+            section_status="hakemusosan tila",
+            section_name="varauksen nimi",
+            reservations_begin_date="hakijan ilmoittaman kauden alkupäivä",
+            reservations_end_date="hakijan ilmoittaman kauden loppupäivä",
+            applied_reservations_per_week="vuoroja, kpl / vko",
+            reservation_min_duration="minimi aika",
+            reservation_max_duration="maksimi aika",
+            reservation_unit_name="varausyksikkö",
+            unit_name="toimipiste",
+            day_of_the_week="toistoviikonpäivä",
+            begin_time="aloitusaika",
+            end_time="päättymisaika",
+            price="hinta",
+        )
+
+    def _get_queryset(self, **kwargs) -> ApplicationSectionQuerySet:
+        return (
+            ApplicationSection.objects.annotate(
+                # Annotate statuses for performance
+                application_status=L("application__status"),
+                status=L("status"),
+            )
+            .filter(
+                application__application_round=self.application_round_id,
+                suitable_time_ranges__isnull=False,
+            )
+            .exclude(
+                application_status__in=[
+                    ApplicationStatusChoice.DRAFT.value,
+                    ApplicationStatusChoice.EXPIRED.value,
+                ],
+            )
+            .select_related(
+                "application",
+                "application__organisation",
+                "application__contact_person",
+                "application__application_round",
+            )
+            .prefetch_related(
+                # All reservation unit options ordered by preferred order
+                Prefetch(
+                    "reservation_unit_options",
+                    ReservationUnitOption.objects.select_related(
+                        "reservation_unit",
+                        "reservation_unit__unit",
+                    ).order_by("preferred_order"),
+                ),
+                Prefetch(
+                    "reservation_unit_options__allocated_time_slots",
+                    AllocatedTimeSlot.objects.alias(
+                        day_of_the_week_number=L("day_of_the_week_number"),
+                    ).order_by(
+                        "day_of_the_week_number",
+                        "begin_time",
+                        "end_time",
+                    ),
+                ),
+            )
+            .order_by("application_id", "id")
+            .distinct()  # Avoid duplicate rows due to implicit joins
+        )
+
+    def _get_single_row_data(self, section: ApplicationSection) -> Iterable[ApplicationSectionExportRow]:
```

Comment:
Return type is inconsistent with base class return type (`BaseCSVExporter._get_single_row_data` returns `list` while this returns an `iterable`, which is a broader type.)

Reply:
Made type hints more consistent, added comments on the base methods 👍